haystack/utils/augment_squad.py

```diff
@@ -0,0 +1,235 @@
+            word_id = glove_word_id_mapping[word]
+            glove_vector = glove_vectors[word_id]
+            with torch.no_grad():
+                word_similarities = torch.mm(glove_vectors, glove_vector.unsqueeze(1)).squeeze(1)
+                ranking = torch.argsort(word_similarities, descending=True)[:word_possibilities + 1]
+                possible_words.append([glove_id_word_mapping[int(id_)] for id_ in ranking.cpu()])
+        else: # word was not in glove either so we can't find any replacements
+            possible_words.append([word])
+
+    return possible_words
+
+def augment(word_id_mapping: dict, id_word_mapping: dict, vectors: np.ndarray, model: PreTrainedModel,
+    tokenizer: PreTrainedTokenizerBase, text: str, multiplication_factor: int = 20, word_possibilities: int = 20,
+    replace_probability: float = 0.4, batch_size: int = 16, device: str = "cpu:0") -> List[str]:
+    # returns a list of different augmented versions of the text
+    replacements = get_replacements(glove_word_id_mapping=word_id_mapping, glove_id_word_mapping=id_word_mapping,
+        glove_vectors=vectors, model=model, tokenizer=tokenizer, text=text, word_possibilities=word_possibilities,
+        batch_size=batch_size, device=device)
+    new_texts = []
+    for i in range(multiplication_factor):
+        new_text = []
+        for possible_words in replacements:
+            if len(possible_words) == 1:
+                new_text.append(possible_words[0])
+                continue
+            if random.random() < replace_probability:
+                new_text.append(random.choice(possible_words[1:]))
+            else:
+                new_text.append(possible_words[0])
+        new_texts.append(" ".join(new_text))
+    return new_texts
+
+def augment_squad(model: str, tokenizer: str, squad_path: Path, output_path: Path,
+        glove_path: Path = Path("glove.txt"), multiplication_factor: int = 20, word_possibilities: int = 20,
+        replace_probability: float = 0.4, device: str = "cpu:0", batch_size: int = 16):
+    """Loads a squad dataset, augments the contexts, and saves the result in SQuAD format."""
+    # loading model and tokenizer
+    transformers_model = AutoModelForMaskedLM.from_pretrained(model)
+    transformers_model.to(device)
+    transformers_tokenizer = AutoTokenizer.from_pretrained(tokenizer, use_fast=False)
+    # load glove for words that do not have one distinct token, but are split into subwords
+    word_id_mapping, id_word_mapping, vectors = load_glove(glove_path=glove_path, device=device)
+
+    # load squad dataset
+    with open(squad_path, "r") as f:
+        squad = json.load(f)
+        
+    topics = []
+
+    for topic in tqdm(squad["data"]):
+        paragraphs = []
+        for paragraph in topic["paragraphs"]:
+            # make every question unanswerable as answer strings will probably match and aren't relevant for distillation
```

Comment:
I am not sure I understand this comment correctly. I understand that answer strings won't be relevant for distillation because we will make predictions with the teacher model anyway. However, what do you mean with "answer strings will probably match"? Why do we want to set `is_impossible` to `True`? That would result in this question being handled as not answerable. Couldn't we leave `question["answers"] = []` as is but have `question["is_impossible"] = False`?